src/appleseed/renderer/kernel/shading/closures.cpp

```diff
@@ -2392,14 +2434,57 @@ void process_transparency_tree(const OSL::ClosureColor* ci, Alpha& alpha)
     alpha.set(1.0f - transparency);
 }
 
-float process_holdout_tree(const OSL::ClosureColor* ci)
+Color3f process_background_tree(const OSL::ClosureColor* ci)
 {
-    return saturate(luminance(do_process_closure_id_tree(ci, HoldoutID)));
+    return do_process_closure_id_tree(ci, BackgroundID);
 }
 
-Color3f process_background_tree(const OSL::ClosureColor* ci)
+bool process_matte_tree(
+    const OSL::ClosureColor*    closure,
+    foundation::Color3f&        matte_color,
+    Alpha&                      matte_alpha)
 {
-    return do_process_closure_id_tree(ci, BackgroundID);
+    if (closure)
+    {
+        switch (closure->id)
+        {
+          case OSL::ClosureColor::MUL:
+            {
+                const OSL::ClosureMul* c = reinterpret_cast<const OSL::ClosureMul*>(closure);
+                process_matte_tree(c->closure, matte_color, matte_alpha);
```

Comment:
We are loosing the result of the process_matte_tree here.
return process_matte_tree(...)